nefertari_sqla/documents.py

```diff
@@ -800,6 +801,39 @@ def get_encoded_field_value(value, _depth, nest_objects=False):
 
         return value
 
+    @classmethod
+    def get_non_indexable_fields(cls):
+        backref_props = get_backref_props(cls)
+
+        def filter_func(f):
+            return not is_indexable(f)
+
+        fields = [column.name for column in filter(filter_func, class_mapper(cls).columns)]
+        fields.extend([prop.key for prop in filter(filter_func, backref_props)])
+
+        return fields
+
+    def get_reverse_non_indexable_fields(self):
+        fields = []
+        related_items = [(getattr(self, item.key), item) for item in get_backref_props(self.__class__)]
+
+        for related_item, relation in related_items:
+            if isinstance(related_item, list):
+                for item in related_item:
+
+                    if relation.back_populates in item.get_non_indexable_fields():
+                        fields.append(relation.key)
```

Comment:
Maybe you could add a `break` here, so that once the `relation.key` is added we leave the inner `for` to continue to the next `relation`